testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -601,6 +602,154 @@ public void testExplainAnalyze()
         assertExplainAnalyze("EXPLAIN ANALYZE SHOW SESSION");
     }
 
+    @Test
+    public void testMaterializedViewsMetadata()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
+            return;
+        }
+
+        String catalogName = getSession().getCatalog().orElseThrow();
+        SchemaTableName materializedView = new SchemaTableName(
+                getSession().getSchema().orElseThrow(),
+                format("test_materialized_view_%s", randomTableSuffix()));
+        SchemaTableName otherMaterializedView = new SchemaTableName(
+                "other_schema",
+                format("test_materialized_view_%s", randomTableSuffix()));
+
+        computeActual(createMaterializedViewSql(materializedView));
+        computeActual(format("CREATE SCHEMA %s", otherMaterializedView.getSchemaName()));
+        computeActual(createMaterializedViewSql(otherMaterializedView));
+
+        // test freshness
+        assertQuery(
+                listMaterializedViewsSql(),
+                getMaterializedViewsResultRows(materializedView, false, otherMaterializedView));
+
+        computeActual(format("REFRESH MATERIALIZED VIEW %s", materializedView));
+
+        assertQuery(
```

Comment:
`BaseConnectorTest` run test in parallel. Hence this test will be clashing with `testMaterializedView` which also creates MVs. You probably use a different schema and narrow down filter (but they you can't test unfiltered case).

I suggest you just add test cases to `testMaterializedView` (below `system.jdbc.columns` assertions)

Reply:
We could just assert contains instead of asserting exactly the specific row?

Reply:
`testMaterializedView` now calls `testMaterializedViewsMetadata`.